src/battery.py

```diff
@@ -0,0 +1,82 @@
+import argparse
+
+import psutil
+
+
+def _get_charging_status():
+    """Get the battery charging status"""
+    battery = psutil.sensors_battery()
+    return battery.power_plugged
+
+
+def get_battery_percent():
+    """Display battery percentage"""
+    if _get_charging_status():
+        return "Charging"
+
+    battery = round(psutil.sensors_battery().percent)
+    return f"{battery}%"
+
+
+def get_battery_time():
+    """Display battery time remaining in hours and minutes"""
+    if _get_charging_status():
+        return "Charging"
+
+    battery = psutil.sensors_battery().secsleft
+    hours, remainder = divmod(battery, 3600)
+    minutes, _ = divmod(remainder, 60)
+    if hours == 0:
+        return f"{minutes}m"
+    return f"{hours}h {minutes}m"
+
+
+def get_battery_human_readable():
+    """Display the remaining battery amount in a fun human-readable format.
+
+    Examples:
+    - Fully charged
+    - Almost full
+    - More than half full
+    ...
+    """
+    if _get_charging_status():
+        return "Charging"
+
+    battery = psutil.sensors_battery().percent
+    if battery == 100:
+        return "Fully charged"
+    elif battery >= 95:
+        return "Almost full"
+    elif battery >= 75:
+        return "More than half full"
+    elif battery == 50:
+        return "Half full"
```

Comment:
I didn't know I could do this, so that's quite cool. The latest implementation is less fun, but uses a match statement which increases readability.

May add these more colorful statements back in later!

Reply:
It is in hindsight arguable if match statements actually improves readability or not. So I may revisit this again!